### PR TITLE
[KernelGen][Nvidia] Add addbmm operator with Triton kernel

### PR DESCRIPTION
**Pull request by LoserCheems**
## Summary
Adds a Triton kernel for `addbmm`. Performs batch matrix-matrix product of matrices with a reduced add step (sum over batch dimension), then adds the bias: `out = beta * input + alpha * (sum_i batch1_i @ batch2_i)`.

## Testing
- Parametrized tests over `MNK_SHAPES` and `FLOAT_DTYPES`
- Validated against PyTorch reference on device side
- Tested on: Nvidia

## Performance
Test command: `pytest benchmark/test_addbmm.py --level core` (NVIDIA H20)

**torch.float16:**

| Shape | Torch (ms) | Gems (ms) | Speedup |
|---|---|---|---|
| [384,384],[2,384,384],[2,384,384] | 0.016 | 0.031 | 0.498x |
| [4096,4096],[2,4096,4096],[2,4096,4096] | 2.103 | 2.122 | 0.991x |
| [1024,1024],[16,1024,1024],[16,1024,1024] | 0.380 | 0.305 | 1.249x |
| [2048,2048],[16,2048,2048],[16,2048,2048] | 2.322 | 2.085 | 1.113x |
| [4096,4096],[16,4096,4096],[16,4096,4096] | 16.643 | 16.653 | 0.999x |
| **Arithmetic Mean** | — | — | **0.970x** |

**torch.float32:**

| Shape | Torch (ms) | Gems (ms) | Speedup |
|---|---|---|---|
| [384,384],[2,384,384],[2,384,384] | 0.033 | 0.031 | 1.079x |
| [4096,4096],[2,4096,4096],[2,4096,4096] | 9.209 | 11.240 | 0.819x |
| [1024,1024],[16,1024,1024],[16,1024,1024] | 1.524 | 1.643 | 0.927x |
| [2048,2048],[16,2048,2048],[16,2048,2048] | 11.349 | 12.620 | 0.899x |
| [4096,4096],[16,4096,4096],[16,4096,4096] | 73.395 | 89.277 | 0.822x |
| **Arithmetic Mean** | — | — | **0.909x** |

**torch.bfloat16:**

| Shape | Torch (ms) | Gems (ms) | Speedup |
|---|---|---|---|
| [384,384],[2,384,384],[2,384,384] | 0.014 | 0.013 | 1.069x |
| [4096,4096],[2,4096,4096],[2,4096,4096] | 2.100 | 1.992 | 1.054x |
| [1024,1024],[16,1024,1024],[16,1024,1024] | 0.375 | 0.303 | 1.235x |
| [2048,2048],[16,2048,2048],[16,2048,2048] | 2.317 | 2.062 | 1.124x |
| [4096,4096],[16,4096,4096],[16,4096,4096] | 16.505 | 15.781 | 1.046x |
| **Arithmetic Mean** | — | — | **1.106x** |

## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS | PASS (15 cases, --level core) | 0.995x | Primary |
| Tianshu | — | — | — | Not tested |
| Muxi | — | — | — | Not tested |
| Ascend | — | — | — | Not tested |
| Hygon | — | — | — | Not tested |

## Files Changed
- `src/flag_gems/ops/addbmm.py`: Triton kernel implementation (fused batch matmul + reduce + bias)
- `tests/test_addbmm.py`: Accuracy test
- `benchmark/test_addbmm.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: LinearAlg, stage: alpha 5.4)

### CODE REVIEW

File: tests/test_addbmm.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import pytest
+import torch
+
+import flag_gems
+
+from . import accuracy_utils as utils
+from .conftest import QUICK_MODE
+
+if QUICK_MODE:
```
Comment (Caeruleann) [inline tests/test_addbmm.py:11]: Add some test shapes.

Comment (LoserCheems) [reply tests/test_addbmm.py:11]: done

**Review by YangLong114514 — APPROVED** [review]

**Review by Caeruleann — APPROVED** [review]
